partner_communication/models/communication_job.py

```diff
@@ -387,12 +393,14 @@ def refresh_text(self, refresh_uid=False):
         for job in self:
             lang = self.env.context.get('lang_preview', job.partner_id.lang)
             if job.email_template_id and job.object_ids:
-                fields = self.env['mail.compose.message'].with_context(
-                    lang=lang).get_generated_fields(
-                    job.email_template_id, [job.id])
+                body = job.report_id.render_qweb_html([job.id])
```

Comment:
I think this is wrong! You are using the Qweb report used for printing the communication and this will produce html with address of the partner and all layout for printing on a paper. However here we are only interested to render the content of the communication. It could as well be that the report is not defined for communications that we never print but only send by email! Please correct this. The goal is to use the mail.template associated with the communication in order to produce the content of the communication (both for the body and the subject.)

Reply:
I wanted to use it but I the method get_generated_fields() but I don't understand how the body_html is generated and the value is always something like: 
**<p>&lt;p&gt;&amp;lt;p&amp;gt;&amp;lt;br&amp;gt;&amp;lt;/p&amp;gt;&lt;/p&gt;</p>**
I am blocked by this

Reply:
The subject is also not generated

Reply:
Like we discussed, it could come from a template error